manifests/charts/istio-control/istio-discovery/files/waypoint.yaml

```diff
@@ -50,7 +50,17 @@ spec:
       terminationGracePeriodSeconds: 2
       serviceAccountName: {{.ServiceAccount | quote}}
       containers:
-      - args:
+      - name: istio-proxy
+        ports:
+        - containerPort: 15020
+          name: waypoint-stats
```

Comment:
should it be http-... ? 

Any naming convention for prometheus ports ? What do we use in sidecars and current gateways ?

In general - I think 'waypoint' is really an internal gateway. We have a tendency to confuse users (and ourselves) with new words like waypoint or hbone - but   it's really a regular gateway using standard http connect protocol with some automated configs.

Reply:
This is what we have for ingress and sidecars: https://github.com/istio/istio/blob/c014d333a18975e73e4ee358684a6b02cd51e473/manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml#L83-L88

I chose these names to attempt to make their function (and the scrape configs) easier to understand for newcomers. Let me think this trough.

Reply:
Why don't we just put that exact thing into waypoint?

```
 - containerPort: 15021 
	   name: status-port 
	   protocol: TCP 
	 - containerPort: 15090 
	   protocol: TCP 
	   name: http-envoy-prom 
```

Reply:
Do we have anything on 15090 for waypoints?

Reply:
Yes ,waypoint is practically identical to ingress

Reply:
One thing I'd like to request is to use the envoy port directly on waypoints. There's no need to go through the agent for stats (they can be O(GB) and choke the go agent).

Reply:
Let me check that. Originally I got the metrics from this endpoint of the waypoint proxies: `http://localhost:15020/stats/prometheus`.

Reply:
15090 appears to be the envoy port that routes to admin port for stats.

Reply:
15090 is direct envoy, 15020 is agent+envoy+app. There is no app for waypoint, but we still do have agent metrics